solana/solana-ibc/programs/solana-ibc/src/chain.rs

```diff
@@ -188,25 +188,25 @@ impl ChainInner {
         trie: &storage::AccountTrie,
         force: bool,
     ) -> Result {
-        let host_head = crate::host::Head::get()?;
+        let (host_height, host_timestamp) = get_host_head()?;
 
         // We attempt generating guest blocks only once per host block.  This
         // has two reasons:
         // 1. We don’t want to repeat the same checks each block.
         // 2. We don’t want a situation where some IBC packets are created
         //    during a Solana block but only some of them end up in a guest
         //    block generated during that block.
-        if self.last_check_height == host_head.height {
+        if self.last_check_height == host_height {
             return if force {
                 Err(Error::GenerationAlreadyAttempted.into())
             } else {
                 Ok(())
             };
         }
-        self.last_check_height = host_head.height;
+        self.last_check_height = host_height;
```

Comment:
As long as change in slot indicates a new block, it should be fine

Reply:
It does. If a slot changes, next leader gets to generate next block.